src/www/ui/api/Controllers/AuthController.php

```diff
@@ -55,7 +55,13 @@ public function getAuthHeaders($request, $response, $args)
     return $response->withHeader('Warning', $warningMessage)->withJson(
       $returnVal->getArray(), $returnVal->getCode());
   }
-
+  public function optionsVerification($request, $response, $args)
+  {
+    $Message = "Successful";
+    $returnVal = new Info(204, $Message, InfoType::INFO);
+    return $response->withJson(
+      $returnVal->getArray(), $returnVal->getCode());
```

Comment:
[Response code 204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) should not have a body. I would suggest following response:
```php
global $SysConf;
return $response->withStatus(204)
  ->withHeader('Access-Control-Allow-Origin', $SysConf['SYSCONFIG']['CorsOrigins'])
  ->withHeader('Access-Control-Allow-Headers', 'X-Requested-With, Content-Type, Accept, Origin, Authorization')
  ->withHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, PATCH, OPTIONS')
  ->withHeader('Access-Control-Allow-Credentials', 'true');
```

Please refer to [this cookbook](https://www.slimframework.com/docs/v3/cookbook/enable-cors.html) for more.

Reply:
And to create the `CorsOrigins` option, please check another change at https://github.com/fossology/fossology/pull/1890/files#diff-24724248fe452296b11be6a8423331269089b3057511f90db0de4babb0b6eb33R395

Reply:
@GMishx sir I have made the suggested changes but I am not getting this CorsOrigins option, I have set up this on my system with this method https://prezi.com/view/dKLarFhLWuQzfPOKpXkr/ please check this out I have documented here, and let me know the changes.

Reply:
@Shruti3004 , did you had a look on the PR link I shared above?

Reply:
Yes, Sir I have checked this pr but i am not able to refer it with cors

Reply:
> @Shruti3004 , did you had a look on the PR link I shared above?

Yes sir, I have checked this pr but I am not able to refer it with cors.

Reply:
I shared the link so you can understand how to create configuration variables in FOSSology. Using that variable, we can simply set the value for `Access-Control-Allow-Origin` header without changing anything in the source code.
You can access the configuration page on FOSSology at Admin > Customize menu.

Reply:
> I shared the link so you can understand how to create configuration variables in FOSSology. Using that variable, we can simply set the value for `Access-Control-Allow-Origin` header without changing anything in the source code.
> You can access the configuration page on FOSSology at Admin > Customize menu.

Okay sir

Reply:
@GMishx as far I get it, what we want to do is we are trying to make the CORS settings dynamic from the user interface so for that I need to add config variable, 
`$variable = "CorsOrigin";`
`$prompt = _('Access-Control-Allow-Origin');`
`$desc = _('[scheme]://[hostname]:[port]');`
`$valueArray[$variable] = array("'$variable'", "null", "'$prompt'",`
    `strval(CONFIG_TYPE_TEXT), "'CorsOrigin'", "1", "'$desc'", "null", "null");`

so sir let me know adding this variable would work or not and should I add localhost:3000 as default because we are going to do it on react

Reply:
Sorry for making it confusing but yes, it does looks good. However, I would propose following changes:
```php
$variable = "CorsOrigins";
$prompt = _('Allowed origins for REST API');
$desc = _('[scheme]://[hostname]:[port], "*" for anywhere');
$valueArray[$variable] = array("'$variable'", "'localhost:3000'", "'$prompt'",
    strval(CONFIG_TYPE_TEXT), "'PAT'", "3", "'$desc'", "null", "null");
```

I just made some changes in the variable name and added the default value `localhost:3000`. Looks good to you?

Reply:
> Sorry for making it confusing but yes, it does looks good. However, I would propose following changes:
> 
> ```
> $variable = "CorsOrigins";
> $prompt = _('Allowed origins for REST API');
> $desc = _('[scheme]://[hostname]:[port], "*" for anywhere');
> $valueArray[$variable] = array("'$variable'", "'localhost:3000'", "'$prompt'",
>     strval(CONFIG_TYPE_TEXT), "'PAT'", "3", "'$desc'", "null", "null");
> ```
> 
> I just made some changes in the variable name and added the default value `localhost:3000`. Looks good to you?

Yes, bhaiya looks good, let me make the following changes.